### PR TITLE
feat: type casting & various fixes

### PR DESCRIPTION
**Pull request by enitrat**
pre-release fixes

### CODE REVIEW

File: src/ch01-03-hello-scarb.md [hunk 1]
```diff
@@ -120,12 +120,16 @@ This command creates a `sierra` file in `target/release`, let's ignore the `sier
...
 
 ```bash
-$ cairo-run -- src/lib.cairo
+$ cairo-run src/lib.cairo
 [DEBUG] Hello, Scarb!                   (raw: 5735816763073854913753904210465)
 
 Run completed successfully, returning []
 ```
 
+> Note: You will notice here a that we didn't use a Scarb command, but rather a command from the Cairo binaries directly.
```
Comment (ClementWalter) [inline src/ch01-03-hello-scarb.md:129]: extra `a`

File: src/ch02-02-data-types.md [hunk 2]
```diff
@@ -18,51 +18,56 @@ fn main(){
...
-A *scalar* type represents a single value. Cairo has three primary scalar types:
+A _scalar_ type represents a single value. Cairo has three primary scalar types:
 felts, integers, and booleans. You may recognize
 these from other programming languages. Let’s jump into how they work in Cairo.
 
 #### Felt Type
 
-In Cairo, if you don't specify the type of a variable or argument, its type defaults to a field element, represented by the keyword `felt252`. In the context of Cairo, when we say “a field element” we mean an integer in the range -P/2 < x < P/2 
- where P is a very large (prime) number (currently it is a 252-bit number, which is a number with 76 decimal digits). When adding, subtracting, or multiplying, if the result falls outside the specified range of the prime number, an overflow occurs, and an appropriate multiple of P is added or subtracted to bring the result back within the range (i.e., the result is computed modulo P).
+In Cairo, if you don't specify the type of a variable or argument, its type defaults to a field element, represented by the keyword `felt252`. In the context of Cairo, when we say “a field element” we mean an integer in the range `0 < x < P`,
```
Comment (ClementWalter) [inline src/ch02-02-data-types.md:29]: `0 <= x < P`

File: src/ch02-02-data-types.md [hunk 3]
```diff
@@ -18,51 +18,56 @@ fn main(){
...
+The most important difference between integers and field elements is division: Division of field elements (and therefore division in Cairo) is unlike regular CPUs division, where
+integer division `x / y` is defined as `[x/y]` where the integer part of the quotient is returned (so you get `7 / 3 = 2`) and it may or may not satisfy the equation `(x / y) *y == x`,
```
Comment (ClementWalter) [inline src/ch02-02-data-types.md:33]: missing space `(x / y) * y = x`

File: src/ch02-02-data-types.md [hunk 4]
```diff
@@ -268,9 +285,9 @@ let mut arr = ArrayTrait::<u128>::new();
 
 ##### Accessing Array Elements
 
-To access array elements, you can use `get()` or `at()` which have different functionning.
+To access array elements, you can use `get()` or `at()` (which is equivalent to the subscripting operator `[]`) Array methods that return different types.
```
Comment (fricoben) [inline src/ch02-02-data-types.md:288]: The `Array methods that return different types.` is a bit strange cause it comes just after the parentheses.

I would do something like: 

`. Two array methods that return different types.`

**Review by fricoben — COMMENTED** [review]
LGTM + Just a small thing to reformulate imo

**Pull request merged**